chunky/src/java/se/llbit/chunky/renderer/renderdump/DumpFormat.java

```diff
@@ -0,0 +1,212 @@
+/* Copyright (c) 2012-2016 Jesper Öqvist <jesper@llbit.se>
+ *
+ * This file is part of Chunky.
+ *
+ * Chunky is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Chunky is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with Chunky.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package se.llbit.chunky.renderer.renderdump;
+
+import se.llbit.chunky.renderer.scene.Scene;
+import se.llbit.util.TaskTracker;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+abstract class DumpFormat {
+  private static final SortedMap<Integer, DumpFormat> DUMP_FORMATS = new TreeMap<>();
```

Comment:
Why not use a list instead and the index in the list as version?

Reply:
That's a good idea! Don't know what my reason for a map was 😅